core/src/verify.rs

```diff
@@ -246,6 +248,28 @@ impl CommitSequenceVerifier {
                 "consensus_delegatee cannot be changed".to_string(),
             ));
         }
+
+        let previous_version = Version::parse(&self.reserved_state.version);
```

Comment:
I think the naming itself can be misleading which could imply that the version **must** be refreshed in an advance way.

Reply:
I agree with your point. The term `previous_version` could indeed be misleading. I'll change it to `current_version` to better reflect its meaning.